packages/shared_preferences/shared_preferences/example/integration_test/shared_preferences_test.dart

```diff
@@ -587,6 +587,17 @@ void main() {
         expect(preferences.getStringList(listKey), testList);
       });
 
+      testWidgets('get StringList handles List<Object>',
+          (WidgetTester _) async {
+        final (
+          SharedPreferencesWithCache preferences,
+          Map<String, Object?> cache
+        ) = await getPreferences();
+        final List<Object> listObject = <Object>['one', 'two'];
```

Comment:
The actual type in the exception was `List<Object?>`, so this should be using that as the type even though none of the elements are null.